csharp/device/Microsoft.Azure.Devices.Client/Common/SHA.cs

```diff
@@ -0,0 +1,211 @@
+// 
+//  Code by Elze Kool
```

Comment:
This is the last open item, everything else looks good. Just to be on the safe side I requested the advice of our opensource licensing/copyright contacts, I'll keep you posted and will be ready to move forward as soon as we get an answer/recommendation.